### PR TITLE
KOA-4637 Update to publish on maven central

### PR DESCRIPTION
**Pull request by marianeum**
This changes our publishing code to publish on maven central, rather than jitpack and an internal artifactory to simplify publishing and reduce publishing failures or inconsistencies.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/gradle-maven-push.gradle [hunk 1]
```diff
@@ -1,82 +1,69 @@
...
+    def props = ["ossrhUsername", "ossrhPassword", "signing.keyId", "signing.password", "signing.secretKeyRingFile", "sonatypeStagingProfileId"]
+    if (props.any { rootProject.ext[it]?.isEmpty() }) {
+      ant.fail("Missing maven credentials")
     }
+  }
 }
+
+ext["signing.keyId"] = rootProject.ext["signing.keyId"]
+ext["signing.password"] = rootProject.ext["signing.password"]
+ext["signing.secretKeyRingFile"] = rootProject.ext["signing.secretKeyRingFile"]
```
Comment (bvitaliyg) [inline Backpack/gradle-maven-push.gradle:64]: Where does this coming from?

File: package.json [hunk 2]
```diff
@@ -18,7 +18,6 @@
     "postemulator": "npm run emulator:kill",
     "prerelease": "npm ci && npm run emulator",
     "release": "node release.js",
-    "postrelease": "node postrelease.js > /dev/null 2>&1 &",
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply package.json:21]: Nope, it was used to trigger build only :)

File: Backpack/gradle-maven-push.gradle [hunk 3]
```diff
@@ -1,82 +1,69 @@
...
+tasks.register('checkMavenCredentials') {
+  doLast {
+    def props = ["ossrhUsername", "ossrhPassword", "signing.keyId", "signing.password", "signing.secretKeyRingFile", "sonatypeStagingProfileId"]
+    if (props.any { rootProject.ext[it]?.isEmpty() }) {
+      ant.fail("Missing maven credentials")
     }
+  }
 }
+
+ext["signing.keyId"] = rootProject.ext["signing.keyId"]
```
[the remark this answers is not in the document]
Comment (marianeum) [reply Backpack/gradle-maven-push.gradle:62]: On maven central you need to sign everything to verify that it was you

File: Backpack/gradle-maven-push.gradle (hunk 1 above)
Comment (marianeum) [reply Backpack/gradle-maven-push.gradle:64]: these come from the local.properties (you can see how these are read from it in the root gradle file) - the details are in lastpass - i'll update the docs as well to include all the updated instructions

File: Backpack/gradle-maven-push.gradle [hunk 4]
```diff
@@ -1,82 +1,68 @@
...
-                artifactId 'backpack-android'
-                version "$version"
-                description description
-
-              artifact("$buildDir/outputs/aar/Backpack-release.aar")
-                artifact sourceJar
+        from components.release
+        artifact sourceJar
 
-                pom.withXml {
```
[the remark this answers is not in the document]
Comment (marianeum) [reply Backpack/gradle-maven-push.gradle:44]: yep docs are still outstanding - wanted to get a first review first before i update those

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -158,7 +158,7 @@ jobs:
             Thank you for your contribution.
             You can use the following dependency to test your application while we review your pull request:
 
-            `com.github.Skyscanner:backpack-android:${{ github.sha }}`
+            `net.skyscanner.backpack:backpack-android:${{ github.sha }}`
```
Comment (marianeum) [inline .github/workflows/ci.yml:161]: how does this actually work? i don't see anything actually doing the publishing?!

File: docs/Toast/README.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 ## Installation
 
-Backpack Android is available through [Jitpack](https://jitpack.io/#Skyscanner/backpack-android). Check the main [Readme](https://github.com/skyscanner/backpack-android#installation) for a complete installation guide.
+Backpack Android is available through [Maven Central](https://search.maven.org/artifact/net.skyscanner.backpack/backpack-android). Check the main [Readme](https://github.com/skyscanner/backpack-android#installation) for a complete installation guide.
```
Comment (bvitaliyg) [inline docs/Toast/README.md:5]: Nice catch! We should probably remove this section eventually or use some sort of snipper for that.

**Review by bvitaliyg — APPROVED** [review]
👍

File: docs/Toast/README.md (hunk 6 above)
Comment (marianeum) [reply docs/Toast/README.md:5]: yeah would be nice if we could essentially merge md files here, rather than having to specify this for each page

**Pull request merged**